_docs/Qabel-Desktop-REST.md

```diff
@@ -0,0 +1,177 @@
+
+# API
+
+The server runs on port 9696 and is started in a thread by the qabel-desktop applications.
+
+## Settings
+
+### Read Settings
+
+* HTTP-Method: GET
+* Ressource: `/settings`
+
+Returns the complete local settings collection as specified in [client configuration](../Qabel-Client-Configuration#local-settings)
+
+#### Return values
+
+|HTTP status code|reason|
+|:----------------:|------|
+| 200 | Settings sent |
+
+### Update Settings
+
+* HTTP-Method: PUT
+* Ressource: `/settings`
+
+Update a single key
+
+| Key | Description |
+| --- | ----------- |
+| key | Key to update, can be in the form of foo.bar.baz |
+| value | Value to set for the key |
+
+#### Return values
+
+|HTTP status code|reason|
+|:----------------:|------|
+| 200 | Setting updated |
+| 400 | value is not valid for this key |
+| 405 | key is not valid |
+
+#### Substitute Settings
+
+* HTTP-Method: POST
+* Ressource: `/settings`
+
+Substitute the current settings.
+
+| Key | Description |
+| --- | ----------- |
+| settings | JSON document of the settings |
+
+#### Return values
+
+|HTTP status code|reason|
+|:----------------:|------|
+| 200 | Setting updated |
+| 400 | Settings contain invalid keys or values |
+| 405 | Settings is not a valid JSON document |
+
+
+## Synced Settings
+
+### Read Settings
+
+* HTTP-Method: GET
+* Ressource: `/synced`
+
+Returns the complete synced settings collection as specified in [client configuration](../Qabel-Client-Configuration#synced-settings)
+
+#### Return values
+
+|HTTP status code|reason|
+|:----------------:|------|
+| 200 | Settings sent |
+
+### Update Settings
+
+* HTTP-Method: PUT
+* Ressource: `/synced`
+
+Update a single key
+
+| Key | Description |
+| --- | ----------- |
+| key | Key to update, can be in the form of foo.bar.baz |
+| value | Value to set for the key |
+
+#### Return values
+
+|HTTP status code|reason|
+|:----------------:|------|
+| 200 | Setting updated |
+| 400 | value is not valid for this key |
+| 405 | key is not valid |
+
+#### Substitute Settings
+
+* HTTP-Method: POST
+* Ressource: `/synced`
+
+Substitute the current settings.
+
+| Key | Description |
+| --- | ----------- |
+| settings | JSON document of the settings |
+
+#### Return values
+
+|HTTP status code|reason|
+|:----------------:|------|
+| 200 | Setting updated |
+| 400 | Settings contain invalid keys or values |
+| 405 | Settings is not a valid JSON document |
+
+## Drops
+
+### Retrieve Drops
+
+* HTTP-Method: GET
+* Ressource: `/drop`
+
+Returns a list of all drop messages from all configured DropURLs.
+See: [Client Drop](../Qabel-Client-Drop]
+
+
+#### Return values
+
+|HTTP status code|reason|
+|:----------------:|------|
+| 200 | Drops sent |
+| 204 | No new drops available |
+
+### Check for drops
+
+* HTTP-Method: HEAD
+* Ressource: `/drop`
+
+Same as retrieving drops, but without sending them.
```

Comment:
Retrieving drops also sends them? Or do I missunderstand the meaning of "send" here?